masonry/src/widgets/slider.rs

```diff
@@ -0,0 +1,845 @@
+                vello::peniko::Fill::NonZero,
+                Affine::IDENTITY,
+                &self.track_color,
+                None,
+                track,
+            );
+        }
+
+        // Draw the thumb
+        let thumb = self.thumb_rect(ctx.size());
+        scene.fill(
+            vello::peniko::Fill::NonZero,
+            Affine::IDENTITY,
+            &self.thumb_color,
+            None,
+            &thumb,
+        );
+
+        if self.is_hovered {
+            scene.draw_blurred_rounded_rect(
+                Affine::IDENTITY,
+                thumb.rect(),
+                self.hover_glow_color,
+                self.thumb_radii.top_left + self.hover_glow_spread_radius,
+                self.hover_glow_blur_radius,
+            );
+        }
+    }
+
+    fn accessibility_role(&self) -> Role {
+        Role::Slider
+    }
+
+    fn accessibility(&mut self, _ctx: &mut AccessCtx, node: &mut Node) {
+        node.set_value(self.value.to_string());
+        node.add_action(accesskit::Action::SetValue);
+    }
+
+    fn children_ids(&self) -> SmallVec<[WidgetId; 16]> {
+        SmallVec::new()
+    }
+
+    fn make_trace_span(&self, ctx: &QueryCtx<'_>) -> Span {
+        trace_span!("Slider", id = ctx.widget_id().trace())
+    }
+
+    fn get_debug_text(&self) -> Option<String> {
+        Some(self.value.to_string())
+    }
+
+    fn get_cursor(&self, ctx: &QueryCtx, _pos: Point) -> CursorIcon {
+        if ctx.is_disabled() {
+            CursorIcon::Wait
+        } else if self.is_dragging {
+            CursorIcon::Grabbing
+        } else if self.is_hovered {
+            CursorIcon::Grab
+        } else {
+            CursorIcon::Default
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::{
+        assert_render_snapshot,
+        core::Action,
+        testing::{widget_ids, TestHarness, TestWidgetExt},
+    };
+    use insta::assert_debug_snapshot;
+
+    const TEST_SIZE: Size = Size::new(200.0, 40.0); // Standard test size
+
+    #[test]
+    fn horizontal_slider_default() {
+        let [slider_id] = widget_ids();
+        let slider = Slider::new(Axis::Horizontal, 0.0, 100.0, 50.0).with_id(slider_id);
+
+        let mut harness = TestHarness::create_with_size(slider, TEST_SIZE);
+
+        let widget = harness
+            .get_widget(slider_id)
+            .downcast::<Slider>()
+            .expect("Slider widget not found");
+        assert_eq!(widget.value, 50.0);
+        assert!(!widget.editing);
+        assert_debug_snapshot!(harness.root_widget());
+        assert_render_snapshot!(harness, "horizontal_slider_default");
+    }
+
+    #[test]
+    fn vertical_slider() {
+        let [slider_id] = widget_ids();
+        let slider = Slider::new(Axis::Vertical, 0.0, 100.0, 75.0).with_id(slider_id);
+
+        let mut harness = TestHarness::create_with_size(slider, Size::new(40.0, 200.0));
+        assert_render_snapshot!(harness, "vertical_slider");
+    }
```

Comment:
It's great that you added lots of screenshot tests, but the screenshots haven't been added to source control.